src/include/sof/audio/component.h

```diff
@@ -862,13 +862,15 @@ void comp_get_copy_limits_with_lock_frame_aligned(struct comp_buffer *source,
 						  struct comp_buffer *sink,
 						  struct comp_copy_limits *cl)
 {
-	source = buffer_acquire(source);
```

Comment:
@andrula-song whats the rationale for using a local variable? I understand it fixes the bug but how do we explain it?

Reply:
The bug is being fixed here as the ordering of the locking and unlocking is now symmetrical.

Reply:
@lgirdwood  But the ordering of the locking and unlocking in function comp_get_copy_limits_with_lock is unsymmetrical too (And that will not lead to the bug). 
Firstly I wanna test without transfer the order, but there would be failed issues of case check-suspend-resume-with-playback-5.sh and check-suspend-resume-with-capture-5.sh on WHL_UPEXT_HDA_ZEPHYR. After transfer the order of the locking and unlocking, these two failure are still exist. But both are not IPC time out. 
@[lyakh](https://github.com/thesofproject/sof/commits?author=lyakh) Can you help to introduce why we should use '__sparse_cache'?

Reply:
> @lgirdwood But the ordering of the locking and unlocking in function comp_get_copy_limits_with_lock is unsymmetrical too (And that will not lead to the bug).

This need fixed too. Pls send a PR. Thanks.